File: test_calculadora.py
import pytest
from calculadora import soma, subtracao, multiplicacao, divisao

def test_soma():
    assert soma(10, 5) == 15
    assert soma(-2, 2) == 0

def test_subtracao():
    assert subtracao(10, 5) == 5

def test_multiplicacao():
    assert multiplicacao(3, 4) == 12

def test_divisao():
    assert divisao(10, 2) == 5

    with pytest.raises(ValueError):
        divisao(10, 0)

File: test_calculadora.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class test_calculadora {

    private final Calculadora calc = new Calculadora();

    @Test
    public void testSoma() {
        assertEquals(5, calc.soma(2, 3));
    }

    @Test
    public void testSubtracao() {
        assertEquals(1, calc.subtracao(4, 3));
    }

    @Test
    public void testMultiplicacao() {
        assertEquals(6, calc.multiplicacao(2, 3));
    }

    @Test
    public void testDivisao() {
        assertEquals(2, calc.divisao(6, 3));
    }
}
